compiler/compile/OMRCompilation.cpp

```diff
@@ -815,14 +815,22 @@ OMR::Compilation::getOSRInductionOffset(TR::Node *node)
    if (osrNode->getOpCode().isCall())
       return 3;
 
-   switch (osrNode->getOpCodeValue())
+   // If the monent has a bytecode index of 0, it must be the
+   // monent for a synchronized method. A transition here
+   // must target the method entry.
+   if (osrNode->getOpCodeValue() == TR::monent)
       {
-      case TR::monent: return 1;
-      case TR::asynccheck: return 0;
-      default:
-         TR_ASSERT(0, "OSR points should only be calls, monents or asyncchecks");
+      if (osrNode->getByteCodeIndex() == 0)
```

Comment:
Is this test sufficient to distinguish between 

the "implicit" monent on method entry for synchronized methods 
vs 
an "explicit" monent that may happen to be the first bytecode (and therefore have bytecode index 0) for a method ?

I believe you were thinking of the former when you implemented this change but I am just wondering if anything could go wrong in the latter case.